### PR TITLE
Cloudfit/removeencryptionathost

### PR DESCRIPTION
**Pull request by coreyperkins**
Removes references to the EncryptionAtHost bits.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,7 +3,6 @@ COMMIT = $(shell git rev-parse --short=7 HEAD)$(shell [[ $$(git status --porcela
 ARO_IMAGE ?= ${RP_IMAGE_ACR}.azurecr.io/aro:$(COMMIT)
 
 ifneq ($(shell uname -s),Darwin)
-    export CGO_CFLAGS=-Dgpgme_off_t=off_t
```
Comment (troy0820) [inline Makefile:6]: This is necessary for people who develop with MacOS

File: pkg/deploy/generator/resources_rp.go [hunk 2]
```diff
@@ -829,7 +831,9 @@ fi
 SECRET_NAME="rp-\${COMPONENT}"
 NEW_CERT_FILE="\$TEMP_DIR/\$COMPONENT.pem"
 for attempt in {1..5}; do
-  az keyvault secret download --file \$NEW_CERT_FILE --id "https://$KEYVAULTPREFIX-svc.vault.azure.net/secrets/\$SECRET_NAME" && break
+  echo "Keyvault Prefix: $KEYVAULTPREFIX"
+  echo "Secret Name: $SECRET_NAME"
```
Comment (troy0820) [inline pkg/deploy/generator/resources_rp.go:835]: Is this necessary to echo?

**Comment by troy0820** [comment]
@coreyperkins Could you squash these commits into one commit?

File: pkg/deploy/generator/resources_rp.go (hunk 2 above)
Comment (jewzaam) [reply pkg/deploy/generator/resources_rp.go:835]: It can be removed, was put in to help debug a deployment.

**Review by troy0820 — DISMISSED** [review]
@coreyperkins Can you squash these commits down to 1 commit?  The changes from the first to the last commit are changes from merges and/or debugging that don't provide any relevant changes that can be squashed.

File: pkg/deploy/generator/resources_rp.go (hunk 2 above)
Comment (troy0820) [reply pkg/deploy/generator/resources_rp.go:835]: Please remove echoing the secret name and keyvault prefix: @coreyperkins

**Comment by jewzaam** [comment]
> Removes references to the EncryptionAtHost bits.

Could you submit with the template description populated? Source for the template in case it didn't populate when you created the PR: https://github.com/Azure/ARO-RP/blob/master/.github/PULL_REQUEST_TEMPLATE.md

**Comment by coreyperkins** [comment]
Which issue this PR addresses:
Removes the EncryptionAtHost boolean that is no longer needed.

What this PR does / why we need it:
Test plan for issue: Standard deployment testing / CI / E2E
Is there any documentation that needs to be updated for this PR? No.

**Comment by coreyperkins** [comment]
> @coreyperkins Can you squash these commits down to 1 commit? The changes from the first to the last commit are changes from merges and/or debugging that don't provide any relevant changes that can be squashed.

@jewzaam could you squash on the merge, Mike said you guys had been down this road before with other PRs.

**Comment by troy0820** [comment]
> could you squash on the merge, Mike said you guys had been down this road before with other PRs.

I can't overwrite your branch.  The squash that happens on merge only removes the merge commit on merge which we keep for history.  

Use an interactive rebase to squash your commits to keep a clean history 
`git rebase -i HEAD~16`.  This will bring you to the screen to pick which commits to keep/discard/squash.  
More information can be found [here](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)

File: pkg/deploy/generator/resources_rp.go [hunk 3]
```diff
@@ -829,7 +831,7 @@ fi
 SECRET_NAME="rp-\${COMPONENT}"
 NEW_CERT_FILE="\$TEMP_DIR/\$COMPONENT.pem"
 for attempt in {1..5}; do
-  az keyvault secret download --file \$NEW_CERT_FILE --id "https://$KEYVAULTPREFIX-svc.vault.azure.net/secrets/\$SECRET_NAME" && break
+  az keyvault secret download --file \$NEW_CERT_FILE --id "https://$KEYVAULTPREFIX-svc.vault.usgovcloudapi.net/secrets/\$SECRET_NAME" && break
```
Comment (mjudeikis) [inline pkg/deploy/generator/resources_rp.go:834]: This does not look right. I expect this should not be here.

File: pkg/deploy/generator/resources_rp.go [hunk 4]
```diff
@@ -519,6 +519,8 @@ cat >/etc/td-agent-bit/td-agent-bit.conf <<'EOF'
 	Port 29230
 EOF
 
+export AZURE_CLOUD_NAME="AzureUSGovernment"
```
Comment (mjudeikis) [inline pkg/deploy/generator/resources_rp.go:522]: This is not the right way to hardcode. Please remove from this PR as this is not the goal of this PR too.

**Review by mjudeikis — CHANGES_REQUESTED** [review]
And please squash. This will ruin all git history

**Review by mjudeikis — DISMISSED** [review]

**Review by mjudeikis — APPROVED** [review]

**Pull request merged**